packages/openactive-integration-tests/test/features/access/access-code-update-notifications/implemented/access-code-update-notifications-test.js

```diff
@@ -0,0 +1,107 @@
+    prerequisite: c1,
+    getInput: () => ({
+      orderItems: fetchOpportunities.getOutput().orderItems,
+    }),
+  });
+  const b = new BFlowStage({
+    ...defaultFlowStageParams,
+    prerequisite: c2,
+    getInput: () => ({
+      orderItems: fetchOpportunities.getOutput().orderItems,
+      totalPaymentDue: c2.getOutput().totalPaymentDue,
+    }),
+  });
+  const [simulateAccessCodeUpdate, orderFeedUpdate] = OrderFeedUpdateFlowStageUtils.wrap({
+    wrappedStageFn: prerequisite => (new TestInterfaceActionFlowStage({
+      ...defaultFlowStageParams,
+      testName: 'Simulate Access Code Update (Test Interface Action)',
+      prerequisite,
+      createActionFn: () => ({
+        type: 'test:AccessCodeUpdateSimulateAction',
+        objectType: 'Order',
+        objectId: b.getOutput().orderId,
+      }),
+    })),
+    orderFeedUpdateParams: {
+      ...defaultFlowStageParams,
+      prerequisite: b,
+      testName: 'Orders Feed (after Simulate Access Code Update)',
+    },
+  });
+
+  // ## Set up tests
+  FlowStageUtils.describeRunAndCheckIsSuccessfulAndValid(fetchOpportunities);
+  FlowStageUtils.describeRunAndCheckIsSuccessfulAndValid(c1);
+  FlowStageUtils.describeRunAndCheckIsSuccessfulAndValid(c2);
+  FlowStageUtils.describeRunAndCheckIsSuccessfulAndValid(b);
+  FlowStageUtils.describeRunAndCheckIsSuccessfulAndValid(simulateAccessCodeUpdate);
+  FlowStageUtils.describeRunAndCheckIsSuccessfulAndValid(orderFeedUpdate, () => {
+    it('should have access codes with altered values', () => {
+      // original = before the AccessCodeUpdateSimulationAction was invoked
+      const originalOrderItems = b.getOutput().httpResponse.body.orderedItem;
+      // new = after the AccessCodeUpdateSimulationAction was invoked
+      const newOrderItems = orderFeedUpdate.getOutput().httpResponse.body.data.orderedItem;
+      // As we'll be setting out expectations in an iteration, this test would
+      // give a false positive if there were no items in `orderedItem`, so we
+      // explicitly test that the OrderItems are present.
+      expect(newOrderItems).to.be.an('array');
+      expect(newOrderItems).to.have.lengthOf.above(0);
+      expect(newOrderItems).to.have.lengthOf(orderItemCriteriaList.length);
+
+      for (const [originalOrderItem, newOrderItem] of zip(originalOrderItems, newOrderItems)) {
+        const originalAccessCodes = originalOrderItem.accessCode;
+        const newAccessCodes = newOrderItem.accessCode;
+
+        // we don't check that newAccessCodes has the same length as originalAccessCodes
+        // as the length could feasibly have changed from the AccessCodeUpdateSimulateAction
+        // All we need to do, therefore, is check that the codes, as a whole, are different.
+        const sortedOriginalAccessCodes = [...originalAccessCodes].sort();
+        const sortedNewAccessCodes = [...newAccessCodes].sort();
+        expect(sortedNewAccessCodes).to.not.deep.equal(sortedOriginalAccessCodes);
+      }
```

Comment:
I've made an assumption here that the access codes are supposed to have actually changed

Reply:
This would then only be testing that the Booking System processes test:AccessCodeUpdateSimulateAction in a way that sort of semantically makes sense (the access codes update to new values). But isn't necessarily testing an internal capability of the Booking System.

So maybe this check is not necessary?